rust/osm2lanes/src/transform/tags_to_lanes/lane_count.rs

```diff
@@ -0,0 +1,375 @@
+use crate::locale::{DrivingSide, Locale};
+use crate::road::Designated;
+use crate::tag::{TagKey, Tags};
+use crate::transform::tags_to_lanes::{Infer, Oneway, RoadBuilder};
+use crate::transform::{RoadWarnings, TagsToLanesMsg};
+
+const LANES: TagKey = TagKey::from("lanes");
+const ONEWAY: TagKey = TagKey::from("oneway");
+
+#[derive(Debug)]
+pub(in crate::transform::tags_to_lanes) struct Counts {
+    pub lanes: Infer<usize>,
+    pub forward: Infer<usize>,
+    pub backward: Infer<usize>,
+    pub both_ways: Infer<usize>,
+}
+
+impl Counts {
+    /// Parses and validates the `lanes` scheme (which excludes parking lanes, bike lanes, etc.).
+    /// See <https://wiki.openstreetmap.org/wiki/Key:lanes>.
+    ///
+    /// Validates `lanes[:{forward,both_ways,backward}]=*` and `centre_turn_lane=yes`.
+    #[allow(clippy::too_many_lines)]
+    pub(in crate::transform::tags_to_lanes) fn new(
+        tags: &Tags,
+        road: &RoadBuilder,
+        locale: &Locale,
+        warnings: &mut RoadWarnings,
+    ) -> Self {
+        // TODO, crosscheck access against tagged
+        let access = Self::get_counts_by_access(tags, warnings);
+
+        // Pre-existing lanes
```

Comment:
I'm a little bit lost without extra context on _when_ this code runs, because it must be after some work has been done. Would it be worth a quick mention in the docstring?